### PR TITLE
support both legacy and SQL syntanx for BigQuery

### PR DESCRIPTION
**Pull request by nevillelyh**
This addresses #216

### CODE REVIEW

File: scio-bigquery/src/main/scala/com/spotify/scio/bigquery/BigQueryClient.scala [hunk 1]
```diff
@@ -390,6 +417,42 @@ class BigQueryClient private (private val projectId: String,
...
+      val jobConfig = new JobConfiguration().setQuery(queryConfig).setDryRun(true)
+      val job = new Job().setConfiguration(jobConfig)
+      bigquery.jobs().insert(projectId, job).execute()
+    }
+    try {
+      run(false)
+      false
+    } catch {
+      case e: GoogleJsonResponseException =>
+        if (e.getDetails.getErrors.get(0).getReason == "invalidQuery") {
```
Comment (ravwojdyla) [inline scio-bigquery/src/main/scala/com/spotify/scio/bigquery/BigQueryClient.scala]: User `errorExtractor`

Comment (nevillelyh) [inline scio-bigquery/src/main/scala/com/spotify/scio/bigquery/BigQueryClient.scala]: Where's that from?

Comment (ravwojdyla) [inline scio-bigquery/src/main/scala/com/spotify/scio/bigquery/BigQueryClient.scala]: `com.google.cloud.hadoop.util.ApiErrorExtractor`

Comment (nevillelyh) [inline scio-bigquery/src/main/scala/com/spotify/scio/bigquery/BigQueryClient.scala]: That class doesn't have a "invalidQuery" reason and `getErrorInfo` is protected :(

**Pull request merged**